File: NC/ArraysHashing/ValidAnagram.go
package main

import "fmt"

/***

 */

func main() {
	fmt.Println(IsValidAnagram("lily", "lele"))
}

func IsValidAnagram(s, t string) bool {

	if len(s) != len(t) {
		return false
	}

	countS := map[string]int{}
	countT := map[string]int{}

	for i := range s {
		countS[string(s[i])]++
		countT[string(t[i])]++
	}

	for i := range countS {
		if countS[i] != countT[i] {
			return false
		}
	}
	return true
}

File: neetcode/array-hashing/valid-anagram/valid-anagram.py


from itertools import count


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        countS, countT = {}, {}

        for i in range(len(s)):
            countS[s[i]] = 1 + countS.get(s[i], 0)   
            countT[s[i]] = 1 + countT.get(s[i], 0)   

        for char in countS:
            if countS[char] != countT.get(char, 0):
                return False
        
        return True